### PR TITLE
Upgrade ed to 1.20

### PR DESCRIPTION
**Pull request by suresh-thelkar**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrade ed to 1.20

###### Change Log  

- NA

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- NA
###### Test Methodology

- Uniform buddy build pipeline resutls are [here](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=501016&view=results)

### CODE REVIEW

File: SPECS/ed/ed.spec [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
 URL:            https://www.gnu.org/software/ed/
 License:        GPLv3
 Group:          Applications/System
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
-# Official source under https://ftp.gnu.org/gnu/ed/ed-1.14.2.tar.lz.
-# We don't have lzip to decompress it.
-Source0:        https://src.fedoraproject.org/repo/pkgs/%{name}/%{name}-%{version}.tar.xz/sha512/de838a6df785c7dc80f4b5ba84330bbe743983fd81218321d4ab84c4c3688fdafb4c005502f3228f0bfa2b6bcf342d64d9523ab73ee440b4f305a033f567cbc2/%{name}-%{version}.tar.xz
-
+# We don't have lzip to decompress it. Hence converted to tar.gz
```
Comment (mfrw) [inline SPECS/ed/ed.spec:10]: 🥇 
Maybe once the churn settles down, we could promote lzip from extended -> core and use upstream sources :)

File: cgmanifest.json [hunk 2]
```diff
@@ -3088,8 +3088,8 @@
         "type": "other",
         "other": {
           "name": "ed",
-          "version": "1.14.2",
-          "downloadUrl": "https://src.fedoraproject.org/repo/pkgs/ed/ed-1.14.2.tar.xz/sha512/de838a6df785c7dc80f4b5ba84330bbe743983fd81218321d4ab84c4c3688fdafb4c005502f3228f0bfa2b6bcf342d64d9523ab73ee440b4f305a033f567cbc2/ed-1.14.2.tar.xz"
+          "version": "1.20",
+          "downloadUrl": "https://ftp.gnu.org/gnu/ed/ed-1.20.tar.lz"
```
Comment (mfrw) [inline cgmanifest.json:3092]: Thank you for the cleanup.

**Review by mfrw — APPROVED** [review]
LGTM :)

**Comment by suresh-thelkar** [comment]
Thank you @mfrw for the review.

**Pull request merged**